app/services/sources/sources.ts

```diff
@@ -616,16 +618,28 @@ export class SourcesService extends StatefulService<ISourcesState> {
     });
   }
 
-  private updateSourceFlags(source: ISource, flags: number, doNotEmit?: boolean) {
+  private updateSourceFlags(source: Source, flags: number, doNotEmit?: boolean) {
```

Comment:
Why `Source` instead of `ISource`?

Reply:
Because `ISource` does not have `getModel` method.